clevertap-core/src/main/java/com/clevertap/android/sdk/cryption/CryptMigrator.kt

```diff
@@ -0,0 +1,292 @@
+                        profile.put(piiKey, migrationResult.data)
+                    }
+                }
+                if (dbAdapter.storeUserProfile(config.accountId, deviceID, profile) <= -1L) {
+                    migrationSuccessful = false
+                }
+            } catch (e: Exception) {
+                config.logger.verbose(config.accountId, "Error migrating profile $deviceID: $e")
+                migrationSuccessful = false
+            }
+        }
+
+        return migrationSuccessful
+    }
+
+
+    /**
+     * This method migrates the encryption level of the inapp data stored in the shared preferences file.
+     * Migration(if needed) is always performed to AES_GCM
+     *
+     * Returns true if migration was successful and false otherwise
+     */
+    private fun migrateInAppData(): Boolean {
+        config.logger.verbose(config.accountId, "Migrating encryption for InAppData")
+        var migrationSuccessful = true
+
+        val sharedPrefsDir = File(context.applicationInfo.dataDir, "shared_prefs")
+
+        // Fetch all SharedPreferences files starting with "inApp" and ending with the accountId
+        val prefsFiles = sharedPrefsDir.listFiles { _, name ->
+            name.startsWith("inApp") && name.endsWith("${config.accountId}.xml")
+        }
+
+        prefsFiles?.forEach { file ->
+            val prefName = file.nameWithoutExtension
+            val prefs = context.getSharedPreferences(prefName, Context.MODE_PRIVATE)
+            val keysToProcess = listOf("inapp_notifs_cs", "inapp_notifs_ss")
+
+            keysToProcess.forEach { key ->
+                prefs.getString(key, null)?.let { data ->
+                    val migrationResult = performMigrationStep(true, data)
+                    migrationSuccessful = migrationSuccessful && migrationResult.migrationSuccessful
+                    prefs.edit().putString(key, migrationResult.data).apply()
+                }
+            }
+        }
+
+        return migrationSuccessful
+    }
+
+    private fun performMigrationStep(encrypt: Boolean, data: String): MigrationResult {
+        val currentState = getCurrentEncryptionState(data)
+        val finalState = getFinalEncryptionState(encrypt)
+
+        if (currentState == finalState) {
+            // No migration needed if current state matches the final state
+            return MigrationResult(data, true)
+        }
+
+        return try {
+            val processedData = currentState.transitionTo(finalState, data, cryptHandler)
+            MigrationResult(data = processedData?:data, migrationSuccessful = processedData != null)
+        } catch (e: Exception) {
+            config.logger.verbose(config.accountId, "Migration step failed for data: $e")
+            MigrationResult(data = data, migrationSuccessful = false)
+        }
+    }
+
+    private fun getFinalEncryptionState(encrypt: Boolean): EncryptionState {
+        return when (encrypt) {
```

Comment:
return if(encrypt) instead of when(encrypt)

Reply:
Addresses in [bab4cbb](https://github.com/CleverTap/clevertap-android-sdk/pull/719/commits/bab4cbbdf777894e45e9521a0c53b5e005d7c6c0)